File: app/src/main/java/com/example/githubrepo/services/GitHubService.java
package com.example.githubrepo.services;

import com.example.githubrepo.models.Repository;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by sarah_neo on 20/02/2017.
 */

public interface GitHubService {

    // 10 calls per minute
    @GET("search/repositories")
    Call<List<Repository>> listRepos(@Query("q") String query,
                                     @Query("sort") String sort,
                                     @Query("order") String order,
                                     @Query("page") int page,
                                     @Query("per_page") int perPage);
}

File: app/src/main/java/com/example/githubrepo/services/GitHubService.kt
package com.example.githubrepo.services

import com.example.githubrepo.models.RepositoryList
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface GitHubService {
    // 10 calls per minute
    @GET("search/repositories")
    fun listRepos(@Query("q") query: String,
                  @Query("sort") sort: String,
                  @Query("order") order: String?,
                  @Query("page") page: Int,
                  @Query("per_page") perPage: Int): Call<RepositoryList>
}